Lib/zipfile/__init__.py

```diff
@@ -331,7 +331,7 @@ def _EndRecData(fpin):
     # number does not appear in the comment.
     maxCommentStart = max(filesize - (1 << 16) - sizeEndCentDir, 0)
     fpin.seek(maxCommentStart, 0)
-    data = fpin.read()
+    data = fpin.read(ZIP_MAX_COMMENT + sizeEndCentDir + 1)
```

Comment:
With the change to line 332 you can remove the `+ 1` and `test.test_zipfile.test_core.OtherTests.test_comments` will still pass.

```suggestion
    data = fpin.read(ZIP_MAX_COMMENT + sizeEndCentDir)
```

If you don't want to make the change to line 332 (as it is a little off topic), I think we should be consistent with how the number of bytes is calculated in line 332:
```suggestion
    data = fpin.read((1 << 16) + sizeEndCentDir)
```

I like how you've done it and my preference would be to fix line 332